### DISCUSSION TITLE
Meta plugin doesn't seem to run in SSR

### DISCUSSION THREAD

**Discussion by gregbengis** [quasarframework/quasar#12669] [CLI - SSR mode]
I've previously used Meta plugin with SSR on some of my websites, with expected results: 
When I look at the source HTML of any page, I see the relevant title, description, etc..

I recently tried to implement the same config on another site of mine, but the Meta data is not rendering SSR.
In other words, SSR is working, and I can build and deploy it, however, the html source contains the default index.template.html title. 

Client side however, after the page loads, the Meta plugin does kick in, and the correct page title appears, etc, so I know I've set it up correctly. Social media and search engines are a problem though.

I upgraded to Quasar v2 and the problem persists.. 

Any ideas? I am not even sure what to additional info to send at this point so I will wait for direction.

Thanks!

**Comment by hawkeye64** [comment]
The Quasar docs run as SSR and it uses the meta plugin. There are no problems there.
Without any more information on your part, I don't think anyone will be able to help you.

**Reply by gregbengis** [reply]
Never suggested anything was wrong with the plugin, just needed assistance in finding a way to a solution, but sorted now.

**Comment by stefanvanherwijnen** [comment]
Are you using the SSG app extension? If not, how do you get html files with SSR?

**Reply by gregbengis** [reply]
Nope, just been using SSR. But sorted now, thanks.

**Comment by hawkeye64** [comment]
@stefanvanherwijnen is on the right track. The meta plugin only works correctly with SSR. See the `tip` here (at top of page): https://quasar.dev/quasar-plugins/meta#introduction

**Reply by gregbengis** [reply]
See solution below, thanks!

**Comment by gregbengis** [comment]
So it was a logic issue. I had a v-if condition (initially set to false) on the main div in my Layout, which was set to true by a setTimeout call, in order to prevent a flicker (which wouldn't have happened SSR side). Should have used CSS.
Anyway, the consequence was that the <router-view tag was not rendered in SSR, only once the the SPA kicked in.

**Reply by stefanvanherwijnen** [reply]
Didn't you get hydration errors when doing that?

**Comment by gregbengis** [comment]
No, strangely no errors in my console .
